parcels/particleset.py

```diff
@@ -48,6 +53,17 @@ def add_jitter(pos, width, min, max):
     return lon, lat
 
 
+def nearest_index(array, value):
+    """returns index of the nearest value in array using O(log n) bisection method"""
+    y = bisect.bisect(array, value)
+    if y == len(array):
+        return y-1
+    elif(abs(array[y-1] - value) < abs(array[y] - value)):
+        return y-1
+    else:
+        return y
```

Comment:
A more fundamental issue: I have seen netcdf output of (global) ocean models where longitude is not monotic. For example, longitude would be [180:360, 0:179] or something.

However, I suspect that other parts of `parcels` would also break on these grids. SO perhaps we should write a wrapper function that checks on read-in of longitude whether it is monotonic, and otherwise adjusts it so that it is (by adding 360 degrees).
So no need to do anything here, but just wanted to flag it now that I'm thinking about it